Docs/markdown syntax (#15861)
gatsbyjs/gatsby: docs/docs/mdx/markdown-syntax.md

---
title: Markdown Syntax
---

### Headings

```
# heading 1
## heading 2
### heading 3
#### heading 4
##### heading 5
###### heading 6
```

# heading 1

## heading 2

### heading 3

#### heading 4

##### heading 5

###### heading 6

- each heading gets converted to their HTML equivalent
  - i.e. `# heading 1` is `<h1>heading 1</h1>`
- Correct usage of each heading should follow the
  [accessibility guidelines](https://www.w3.org/WAI/tutorials/page-structure/headings/) set by the World Wide Web Consortium (W3C)

### Emphasized text

- Italic
  - one asterisk or one underscore
    - `*italic*` or `_italic_`
    - _italic!_
- Bold
  - two asterisks or two underscores
    - `**bold**` or `__bold__`
    - **bold!**
- Italic and Bold

  - three asterisks or three underscore
    - `***italic and bold***` or `___italic and bold___`
    - **_italic and bold!!_**

> ### Lists

Reviewer: ```suggestion
## Lists
```

> - Unordered

Reviewer: ```suggestion
### Unordered
```
Recommending a heading here since the list format is a little _too_ meta when talking about lists! 🤯

>   - can use `*, -, +`

Reviewer: ```suggestion
  - can use `*`, `-`, or `+` for each list item
```

  ```
  * Gatsby
    * docs
  - Gatsby
    - docs
  + Gatsby
    + docs
  ```

> 

Reviewer: ```suggestion
How unordered lists are rendered in HTML:
```

  - Gatsby
    - docs

  * Gatsby
    - docs

  - Gatsby
    - docs

> - Ordered

Reviewer: ```suggestion
### Ordered
```

>   - use number and period

Reviewer: ```suggestion
  - number and period for each list item
  - using `1.` for each item can automatically increment depending on the content 
```
Should we add that tip? e.g.:
```
1. One
1. Two
1. Three
```

  ```
    1. One
    2. Two
    3. Three
  ```
  1. One
  2. Two
  3. Three

> ### Links and Images

Reviewer: ```suggestion
## Links and Images
```

> - Link

Reviewer: ```suggestion
### Link
```

> 

Reviewer: ```suggestion
Links in Markdown use this format. URLs can be relative or remote:
```

```
[Text](url)
```

> 

Reviewer: ```suggestion
Example of a link rendering in HTML:
```

[Gatsby site](https://www.gatsbyjs.org/)

> - Image with alt text

Reviewer: ```suggestion
### Image with alt text
```

```
![alt text](path-to-image)
```

> - Image without alt text

Reviewer: ```suggestion
### Image without alt text

This pattern is appropriate for [decorative or repetitive images](https://www.w3.org/WAI/tutorials/images/decision-tree/):
```

```
![](path-to-image)
```

> ### Blockquote

Reviewer: ```suggestion
## Blockquote
```

- Use `>` to declare a blockquote
- Adding multiple `>` with create nested blockquotes
- It is recommended to place `>` before each line

> - You can use other markdown syntax inside blockquotes

Reviewer: ```suggestion
- You can use other Markdown syntax inside blockquotes
```

```
> blockquote
>
> > nested blockquote
>
> > **I'm bold!**
>
> more quotes
```

> Blockquote
>
> > nested blockquote
>
> > **I'm bold!**
>
> more quotes

## Code Comments

> - Inline:

Reviewer: ```suggestion
### Inline
```

  - Enclose the text in backticks \`code\`

>   - Your `code` will look like this

Reviewer: ```suggestion
  - Inline `code` looks like this sentence
```

> * Code Blocks:

Reviewer: ```suggestion
### Code Blocks
```

  - Indent a block by four spaces

>   -     Your code will look like this

Reviewer: ```suggestion
    Your code will look like this
```
I think the list item is a bit confusing here, we could remove it?

## MD vs MDX

- MDX is a superset of Markdown. It allows you to write JSX inside markdown. This includes importing and rendering React components!

## Processing Markdown and MDX in Gatsby:

> - In order to process and use Markdown or MDX in Gatsby, you can use the [gatsby-source-filesystem](https://www.gatsbyjs.org/docs/sourcing-from-the-filesystem) plugin

Reviewer: ```suggestion
- In order to process and use Markdown or MDX in Gatsby, you can use the [gatsby-source-filesystem](/docs/sourcing-from-the-filesystem) plugin
```

> - You can check out the package [here](https://www.gatsbyjs.org/packages/gatsby-source-filesystem) for more information on how it works!

Reviewer: ```suggestion
- You can check out the package [README](/packages/gatsby-source-filesystem) for more information on how it works!
```

## Frontmatter

> - Metadata for your markdown

Reviewer: ```suggestion
- Metadata for your Markdown
```
